File: libsrc/pylith/materials/RheologyThermoelasticity.cc
// =================================================================================================
// This code is part of PyLith, developed through the Computational Infrastructure
// for Geodynamics (https://github.com/geodynamics/pylith).
//
// Copyright (c) 2010-2025, University of California, Davis and the PyLith Development Team.
// All rights reserved.
//
// See https://mit-license.org/ and LICENSE.md and for license information.
// =================================================================================================

#include <portinfo>

#include "pylith/materials/RheologyThermoelasticity.hh" // implementation of object methods

#include "pylith/utils/error.hh" // USES PYLITH_METHOD_*
#include "pylith/utils/journals.hh" // USES PYLITH_JOURNAL_*

// ---------------------------------------------------------------------------------------------------------------------
// Default constructor.
pylith::materials::RheologyThermoelasticity::RheologyThermoelasticity(void) {
    PyreComponent::setName("rheologythermoelasticity");
} // constructor


// ---------------------------------------------------------------------------------------------------------------------
// Destructor.
pylith::materials::RheologyThermoelasticity::~RheologyThermoelasticity(void) {
    deallocate();
} // destructor


// ---------------------------------------------------------------------------------------------------------------------
// Deallocate PETSc and local data structures.
void
pylith::materials::RheologyThermoelasticity::deallocate(void) {
} // deallocate


// ---------------------------------------------------------------------------------------------------------------------
// Update kernel constants.
void
pylith::materials::RheologyThermoelasticity::updateKernelConstants(pylith::real_array* kernelConstants,
                                                                   const PylithReal dt) const {
    // Default is to do nothing.
} // updateKernelConstants


// End of file

File: pylith/materials/RheologyThermoelasticity.py
# =================================================================================================
# This code is part of PyLith, developed through the Computational Infrastructure
# for Geodynamics (https://github.com/geodynamics/pylith).
#
# Copyright (c) 2010-2025, University of California, Davis and the PyLith Development Team.
# All rights reserved.
#
# See https://mit-license.org/ and LICENSE.md and for license information. 
# =================================================================================================

from pylith.utils.PetscComponent import PetscComponent


class RheologyThermoelasticity(PetscComponent):
    """
    Abstract base class for bulk rheology for thermoelasticity.
    """

    import pythia.pyre.inventory

    from pylith.topology.Subfield import subfieldFactory
    from pylith.utils.EmptyBin import EmptyBin

    auxiliarySubfields = pythia.pyre.inventory.facilityArray("auxiliary_subfields", itemFactory=subfieldFactory, factory=EmptyBin)
    auxiliarySubfields.meta['tip'] = "Discretization information for physical properties and state variables."

    def __init__(self, name="rheologythermoelasticity"):
        """Constructor.
        """
        PetscComponent.__init__(self, name, facility="thermoelasticity_rheology")

    def preinitialize(self, problem):
        """Setup rheology.
        """
        self._createModuleObj()

    def addAuxiliarySubfields(self, physics, problem):
        """Add subfields for rheology to auxiliary field.
        """
        return

    def _createModuleObj(self):
        """Call constructor for module object for access to C++ object.
        """
        raise NotImplementedError("Implement in derived class.")


# End of file
